server/api/node_test.go

```diff
@@ -1,22 +1,284 @@
+	node.Name = "Node2"
+	node.Description = "Description2"
+	createdNode2, resp2, err2 := th.AdminClient.CreateNode(&node)
+	require.NoError(t, err2)
+	functionaltesting.CheckCreatedStatus(t, resp2)
+	t.Run("can delete node", func(t *testing.T) {
+		resp, err := th.AdminClient.DeleteNode(createdNode.ID)
+		require.NoError(t, err)
+		functionaltesting.CheckOKStatus(t, resp)
+		deletedNode, rest, err2 := th.AdminClient.GetNode(createdNode.ID)
+		require.NoError(t, err2)
+		functionaltesting.CheckOKStatus(t, rest)
+		require.Truef(t, deletedNode.DeletedAt > 0, "node = %v", deletedNode)
+	})
+
+	t.Run("should not  delete node twice", func(t *testing.T) {
+		resp, err := th.AdminClient.DeleteNode(createdNode2.ID)
+		require.NoError(t, err)
+		functionaltesting.CheckOKStatus(t, resp)
+
+		resp, err = th.AdminClient.DeleteNode(createdNode2.ID)
+		require.NoError(t, err)
+		functionaltesting.CheckOKStatus(t, resp)
+	})
+	t.Run("can not delete non existing node", func(t *testing.T) {
+		resp, err := th.AdminClient.DeleteNode("someInvalidID")
+		require.Error(t, err)
+		functionaltesting.CheckBadRequestStatus(t, resp)
+	})
+}
+
+func TestGetNode(t *testing.T) {
+	th := functionaltesting.Setup(t)
+	defer th.TearDown()
+	node := testNode
+	node.Name = "Node1"
+	node.Description = "Description1"
+	createdNode, resp, err := th.AdminClient.CreateNode(&node)
+	require.NoError(t, err)
+	functionaltesting.CheckCreatedStatus(t, resp)
+	t.Run("can get node", func(t *testing.T) {
+		node, resp, err := th.AdminClient.GetNode(createdNode.ID)
+		require.NoError(t, err)
+		functionaltesting.CheckOKStatus(t, resp)
+		require.Truef(t, node.Name == "Node1", "node = %v", node)
+	})
+
+	t.Run("can't get node with invalid node_id", func(t *testing.T) {
+		_, resp, err := th.AdminClient.GetNode("invalidnodeid")
+		require.Error(t, err)
+		functionaltesting.CheckBadRequestStatus(t, resp)
+	})
+}
+
+func TestGetNodesWithPagination(t *testing.T) {
+	th := functionaltesting.Setup(t)
+	defer th.TearDown()
+
+	// Create a large number of nodes
+	for i := 0; i < 50; i++ {
+		node := testNode
+		node.Name = "Node" + strconv.Itoa(i)
+		_, _, err := th.AdminClient.CreateNode(&node)
+		require.NoError(t, err)
 	}
-	_, resp, err := th.Client.CreateNode(&node)
-	require.NotNil(t, err)
-	functionaltesting.CheckUnauthorizedStatus(t, resp)
+
+	t.Run("can get first page of nodes", func(t *testing.T) {
+		nodes, _, err := th.AdminClient.GetNodes(0, 10, "", "", "")
+		require.NoError(t, err)
+		require.Len(t, nodes, 10)
+	})
+
+	t.Run("can get last page of nodes", func(t *testing.T) {
+		nodes, _, err := th.AdminClient.GetNodes(4, 10, "", "", "")
+		require.NoError(t, err)
+		require.Len(t, nodes, 10)
+	})
+
+	t.Run("can get middle page of nodes", func(t *testing.T) {
+		nodes, _, err := th.AdminClient.GetNodes(2, 10, "", "", "")
+		require.NoError(t, err)
+		require.Len(t, nodes, 10)
+	})
 }
+
+// will be implemented in the future. Needs status field in the NodeWithResources struct
```

Comment:
Should we remove this comment?

Reply:
That's right. I'll add a proper test case here. Thanks